drivers/spi/spi_sc18is606.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2025, tinyvision.ai
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT nxp_sc18is606_spi
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_sc18is606, CONFIG_SPI_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <stdint.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/spi.h>
+#include "spi_context.h"
+
+#define SC18IS606_CONFIG_SPI 0xF0
+#define CLEAR_INTERRUPT      0xF1
+#define IDLE_MODE            0xF2
+
+#include "spi_sc18is606.h"
+
+struct nxp_sc18is606_data {
+	struct k_mutex bridge_lock;
+	struct spi_context ctx;
+	uint32_t frequency;
+	uint8_t spi_mode;
+};
+
+struct nxp_sc18is606_config {
+	const struct i2c_dt_spec i2c_controller;
+};
+
+int nxp_sc18is606_claim(const struct device *dev)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+
+	return k_mutex_lock(&data->bridge_lock, K_FOREVER);
+}
+
+int nxp_sc18is606_release(const struct device *dev)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+
+	return k_mutex_unlock(&data->bridge_lock);
+}
+
+int nxp_sc18is606_transfer(const struct device *dev, const uint8_t *tx_data, uint8_t tx_len,
+			   uint8_t *rx_data, uint8_t rx_len, uint8_t *id_buf)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+	const struct nxp_sc18is606_config *info = dev->config;
+	int ret;
+
+	ret = k_mutex_lock(&data->bridge_lock, K_FOREVER);
```

Comment:
you have many return paths (on I2C transfer error) in this function where this mutex is not being released

Reply:
Thank you so much for catch I will deal with this, Is this still relevant given I have a deadline for this Mutex?

Reply:
Aaaah I see, took me a minute, the fail paths. Sorry about this. fixing